From 47429e341c0097c9b8f04a67e84a28df5781f2d3 Mon Sep 17 00:00:00 2001
From: Clement Calmels <clement.calmels@intel.com>
Date: Wed, 20 Mar 2013 17:29:21 +0100
Subject: [PORT FROM MAIN] Trace when reaching the maximum number of property

BZ: 94604

As seen with bz 80168 this trace can save time during analyse.

This patch adds a trace when the maximum number of property is
reached. In this case, init will also dump the content of the property
area for debugging purpose.

Change-Id: Ic8327dcdfabfe7576e796915c377f7a70c06f589
Orig-Change-Id: Iff30cefa4c09be1ee8be5b148c63fa3864c2c83a
Signed-off-by: Clement Calmels <clement.calmels@intel.com>
Reviewed-on: http://android.intel.com:8080/97747
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Du, Chenyang <chenyang.du@intel.com>
Reviewed-by: Chen, HuaqiangX <huaqiangx.chen@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Tested-by: Chen, HuaqiangX <huaqiangx.chen@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 init/property_service.c |   13 ++++++++++++-
 1 files changed, 12 insertions(+), 1 deletions(-)

diff --git a/init/property_service.c b/init/property_service.c
index 32b77ac..9315d22 100755
--- a/init/property_service.c
+++ b/init/property_service.c
@@ -361,7 +361,18 @@ int property_set(const char *name, const char *value)
         __futex_wake(&pa->serial, INT32_MAX);
     } else {
         pa = __system_property_area__;
-        if(pa->count == PA_COUNT_MAX) return -1;
+        if(pa->count == PA_COUNT_MAX) {
+            int i;
+            ERROR("sys_prop: unable to set property %s: maximum number of property reached\n", name);
+
+            ERROR("sys_prop: properties dump:\n");
+            for (i = 0; i < PA_COUNT_MAX; i++) {
+                pi = TOC_TO_INFO(pa, pa->toc[i]);
+
+                ERROR("sys_prop:\t[%s]: [%s]\n", pi->name, pi->value);
+            }
+            return -1;
+        }
 
         pi = pa_info_array + pa->count;
         pi->serial = (valuelen << 24);
-- 
1.7.4.1

